### PR TITLE
rocAL - HIP fix

### PR DESCRIPTION
**Pull request by rrawther**
fix bug in releasing HIP buffers.

### CODE REVIEW

File: amd_openvx_extensions/amd_media/encoder.cpp [hunk 1]
```diff
@@ -514,7 +514,7 @@ static vx_status VX_CALLBACK amd_media_encode_gpu_buffer_update_callback(vx_node
     if (!encoder) return VX_FAILURE;
 
     return encoder->UpdateBufferOpenCL((vx_image)parameters[1], (vx_array)parameters[2]);
-    return VX_SUCCESS;
+    //return VX_SUCCESS;
```
Comment (asalmanp) [inline amd_openvx_extensions/amd_media/encoder.cpp:517]: can you please remove this line as it is commented out?

File: rocAL/rocAL/include/circular_buffer.h [hunk 2]
```diff
@@ -93,8 +94,8 @@ class CircularBuffer
     cl_device_id _device_id = nullptr;
 #else
     hipStream_t _hip_stream;
-    int _hip_device_id;
-    hipDeviceProp_t *_dev_prop;
+    int _hip_device_id, _hip_canMapHostMemory;
+    //hipDeviceProp_t _dev_prop;
```
Comment (asalmanp) [inline rocAL/rocAL/include/circular_buffer.h:98]: please remove the commented-out line here as well.

**Comment by paveltc** [comment]
@kiritigowda This PR passes the rocAL unit tests.

**Comment by rrawther** [comment]
@kiritigowda : Do you know why it is failing on the package build?

**Comment by kiritigowda** [comment]
OpenCL failure 

```
 [ 70%] Building CXX object rocAL/rocAL/CMakeFiles/rali.dir/source/ring_buffer.cpp.o
/var/jenkins_home/workspace/g_Short-GPUOpen_MIVisionX_PR-730/eGowAXlNN/mivisionx/rocAL/rocAL/source/ring_buffer.cpp:238:30: error: no 'void RingBuffer::release_hip()' member function declared in class 'RingBuffer'
 void RingBuffer::release_hip()
                              ^
rocAL/rocAL/CMakeFiles/rali.dir/build.make:1336: recipe for target 'rocAL/rocAL/CMakeFiles/rali.dir/source/ring_buffer.cpp.o' failed
make[2]: *** [rocAL/rocAL/CMakeFiles/rali.dir/source/ring_buffer.cpp.o] Error 1
CMakeFiles/Makefile2:743: recipe for target 'rocAL/rocAL/CMakeFiles/rali.dir/all' failed
make[1]: *** [rocAL/rocAL/CMakeFiles/rali.dir/all] Error 2
Makefile:171: recipe for target 'all' failed
make: *** [all] Error 2
```

File: amd_openvx_extensions/amd_media/encoder.cpp (hunk 1 above)
Comment (rrawther) [reply amd_openvx_extensions/amd_media/encoder.cpp]: done

File: rocAL/rocAL/include/circular_buffer.h (hunk 2 above)
Comment (rrawther) [reply rocAL/rocAL/include/circular_buffer.h]: done

**Pull request merged**